fetch-prebuilt-helper.js

```diff
@@ -0,0 +1,129 @@
+  
+  TODO: evaluate falling back to https://github.com/chemdrew/npm-golang`)
+  process.exit(1)
+}
+
+const found = whereis('systrayhelper')
+if (found !== '') {
+  console.warn('systrayhelper already installed.')
+  testExecutable(found)
+  process.exit(0)
+}
+
+const tmpDownload = join(os.tmpdir(), 'node-systray-downloadHelper')
+const tmpUnpack = join(os.tmpdir(), 'node-systray-unpack')
+
+console.warn('systrayhelper not installed!')
+console.log('trying to fetching prebuilt for:', process.platform)
+console.log('donload location:', tmpDownload)
+
+try {
+  const urls = {
+    'win32': {
+      'x84': 'https://github.com/ssbc/systrayhelper/releases/download/v0.0.2/systrayhelper_0.0.2_windows_amd64.zip',
+      'ia32': 'https://github.com/ssbc/systrayhelper/releases/download/v0.0.2/systrayhelper_0.0.2_windows_386.zip'
+    },
+    'linux': {
+      'x64': 'https://github.com/ssbc/systrayhelper/releases/download/v0.0.2/systrayhelper_0.0.2_linux_amd64.tar.gz'
+    },
+    'darwin': {
+      'x64': 'https://github.com/ssbc/systrayhelper/releases/download/v0.0.2/systrayhelper_0.0.2_darwin_amd64.tar.gz'
+    }
+  }
+  const hasOS = urls[process.platform]
+  if (!hasOS) {
+    throw new Error('unsupported platform:' + process.platform)
+  }
+  const fileUrl = hasOS[process.arch]
+  if (!fileUrl) {
+    throw new Error('unsupported architecture:' + process.arch)
+  }
+
+  // TODO: find out how to pipe through this:
+  // const ProgressBar = require('progress')
+  // var bar = new ProgressBar('  downloading [:bar] :rate/bps :percent :etas', {
+  //   complete: '=',
+  //   incomplete: ' ',
+  //   width: 20,
+  //   total: total
+  // })
+
+  // unpackaPhase
+  if (process.platform === 'win32') {
+    request
+      .get(fileUrl)
+      .on('error', errorAndExit)
+      .pipe(fs.createWriteStream(tmpDownload))
+      .on('finish', () => {
+        console.log('finished dowloading')
+        const zip = new admZip(tmpDownload)
+        console.log('start unzip')
+        zip.extractAllTo(tmpUnpack, true)
+        console.log('finished unzip')
+        p = join(tmpUnpack, 'systrayhelper.exe')
+        testExecutable(p)
+        cleanup(p)
+      })
+  } else {
+    request
+      .get(fileUrl)
+      .on('error', errorAndExit)
+      .pipe(gunzip())
+      .pipe(tarFs.extract(tmpUnpack))
+      .on('finish', () => {
+        console.log('finished untar')
+        const p = join(tmpUnpack, 'systrayhelper')
+        testExecutable(p)
+        cleanup(p)
+      })
+  }
+} catch (e) {
+  errorAndExit(e)
+}
+
+function testExecutable (path) {
+  console.log('testing execution')
+  try {
+    execSync(path + ' --test')
+    // if (exitCode !== 0) {
+    //   errorAndExit(new Error('unexpected exit from helper. exitCode:' + exitCode))
+    // }
+  } catch (e) {
+    errorAndExit(e)
+  }
+  console.log('helper started succesful!')
+}
+
+function cleanup (path) {
+  fs.renameSync(path, helperLocation)
+  fs.chmodSync(helperLocation, 'u+x')
```

Comment:
This is wrong. It needs to be the octal literal or string representation of the octal bits for the permission you want.

[8.11 docs](https://nodejs.org/dist/latest-v8.x/docs/api/fs.html#fs_file_modes)
`fs.chmodSync(helperLocation, "100");`